server/src/main/java/org/elasticsearch/rest/RestRequestFilter.java

```diff
@@ -28,50 +34,185 @@
 public interface RestRequestFilter {
```

Comment:
why does this class need to change ?

Reply:
Good question!
Short answer is that this filtering, which is needed for auditing, currently works on `RestRequest` types, but we now need it to work on HTTP request bodies.

And that's because auditing works with HTTP requests, because authentication now has to work with HTTP requests, and the two are very much entangled.

BUT I have a proposal to decouple auditing for authentication success from the actual authentication (which sounds bad, I admit) and this is the main reason: today, when we only support authentication on the "REST" channel (no transport client), and the authentication happens before the request body is available, and we want to audit the request body, we cannot cleanly couple authentication and auditing.

Reply:
Basically, it has to do with auditing, which is not properly addressed by this PR anyway (authn success is audited but not the request body), and for which I have a different proposal anyway (that will not require such changes).